### PR TITLE
Introduce pay gate for correlation analysis

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

image

Pay gate, like for Paths! 

## How did you test this code?

toggle enterprise license & check things work.

### CODE REVIEW

File: frontend/src/scenes/project/Settings/index.tsx [hunk 1]
```diff
@@ -189,7 +189,8 @@ export function ProjectSettings(): JSX.Element {
                     you apply a Cohort filter, it means toggling filtering on will match only this specific cohort.
                 </p>
                 <TestAccountFiltersConfig />
-                {featureFlagLogic.values.featureFlags[FEATURE_FLAGS.CORRELATION_ANALYSIS] ? (
+                {featureFlagLogic.values.featureFlags[FEATURE_FLAGS.CORRELATION_ANALYSIS] &&
```
Comment (hazzadous) [inline frontend/src/scenes/project/Settings/index.tsx:192]: oops

**Review by hazzadous — APPROVED** [review]

File: frontend/src/scenes/project/Settings/index.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/project/Settings/index.tsx:192]: Introduced in this PR, so all good 😂

**Comment by hazzadous** [comment]
non-blocking, just curious

What do the server side changes actually do? An do we need to also put in permissioning at the API level?

**Comment by neilkakkar** [comment]
server side: Introduces the feature `CORRELATION_ANALYSIS` as a part of `SCALE` features. These are queried mostly on the frontend when checking if this specific feature is available.

So far, we've opted for not restricting API calls based on enabled features, but that does sound reasonable. Will do this in a separate PR.

File: frontend/src/scenes/insights/FunnelCorrelation.tsx [hunk 2]
```diff
@@ -34,6 +37,16 @@ export const FunnelCorrelation = (): JSX.Element | null => {
         return null
     }
 
+    if (!correlationAnalysisAvailable) {
+        return (
+            <PayCard
+                identifier={AvailableFeature.CORRELATION_ANALYSIS}
+                title="Get a deeper understanding of what your users are doing"
+                caption="Correlation Analysis allows you to find signals for conversions and dropoffs that help you gain deeper insights."
```
Comment (marcushyett-ph) [inline frontend/src/scenes/insights/FunnelCorrelation.tsx:45]:
```suggestion
                caption="Correlation Analysis automates analysis over your events to find signals for why users are converting or dropping off."
```

Comment (marcushyett-ph) [reply frontend/src/scenes/insights/FunnelCorrelation.tsx:45]: Wanted to try and make it clearer - "deeper insights" feels a little vague

Comment (neilkakkar) [reply frontend/src/scenes/insights/FunnelCorrelation.tsx]: Yes, I like this. (I spent literally 3 seconds thinking about the copy)

**Review by paolodamico — APPROVED** [review]

**Pull request merged**